File: apps/net/Ch7_Db/Services/CatalogService.cs
using System.Linq;
using Ch7_Db.Data;

namespace Ch7_Db.Services
{
    public static class CatalogService
    {
        public static Guitar[] AllGuitars(string style = null)
        {
            using (var db = new AppDbContext())
            {
                if (style == null || style == "all")
                {
                    return db.Guitars.OrderByDescending(g => g.Price).ToArray();
                }

                return (from g in db.Guitars
                        where g.Style == style
                        orderby g.Price descending
                        select g
                    ).ToArray();

            }
        }
    }
}

File: apps/py/ch08_db/guitary/services/catalog_service.py
from typing import List, Optional

from guitary.data.guitar import Guitar
from guitary.data import session_factory


def all_guitars(style: Optional[str]) -> List[Guitar]:
    with session_factory.create_session() as ctx:
        if style is None or style == 'all':
            guitars = ctx.session.query(Guitar).order_by(Guitar.price.desc())
            return list(guitars)

    filtered_guitars = list(ctx.session.query(Guitar)
                            .filter(Guitar.style == style)
                            .order_by(Guitar.price.desc()))

    return filtered_guitars
